shared/js/blobview.js

```diff
@@ -420,33 +333,36 @@ var BlobView = (function() {
     },
 
     // Read bytes up to and including a null terminator, but never
-    // more than size bytes.  And return as a Latin1 string
-    readNullTerminatedLatin1Text: function(size) {
-      var s = '';
-      for (var i = 0; i < size; i++) {
-        var charcode = this.view.getUint8(this.index + i);
-        if (charcode === 0) {
-          i++;
-          break;
-        }
-        s += String.fromCharCode(charcode);
+    // more than size bytes.  And return as an ASCII string
+    readNullTerminatedASCIIText: function(size) {
+      var bytes = new Uint8Array(this.buffer, this.viewOffset + this.index,
+                                 size);
+
+      var nil = bytes.indexOf(0);
+      if (nil !== -1) {
+        bytes = bytes.subarray(0, nil);
       }
-      this.index += i;
+
+      var decoder = new TextDecoder('ascii');
+      var s = decoder.decode(bytes);
+      this.index += (nil !== -1) ? nil + 1 : size;
```

Comment:
Nevermind. Your code is correctly. bytes.length doesn't include the nul-terminator. Sorry.